web/src/features/comments/validateCommentReferenceObject.ts

```diff
@@ -39,27 +48,42 @@ export const validateCommentReferenceObject = async ({
   input: z.infer<typeof CreateCommentData>;
 }): Promise<{ errorMessage?: string }> => {
   const { objectId, objectType, projectId } = input;
-  const prismaModel = COMMENT_OBJECT_TYPE_TO_PRISMA_MODEL[objectType];
 
-  if (!prismaModel) {
-    return {
-      errorMessage: `No prisma model for object type ${objectType}`,
-    };
-  }
+  if (env.CLICKHOUSE_URL && isObservationOrTrace(objectType)) {
+    let clickhouseObject;
+    if (objectType === CommentObjectType.OBSERVATION) {
+      clickhouseObject = await getObservationById(objectId, projectId);
+    } else {
+      clickhouseObject = await getTraceById(objectId, projectId);
+    }
 
-  const model = ctx.prisma[prismaModel];
-  const object = await model.findFirst({
-    where: {
-      id: objectId,
-      projectId,
-    },
-  });
+    return !!clickhouseObject
+      ? {}
+      : {
+          errorMessage: `Reference object, ${objectType}: ${objectId} not found in Clickhouse. Skipping creating comment.`,
+        };
+  } else {
```

Comment:
If one of our clickhouse experts could quickly review this please, that would be very helpful. Just want to make sure my implementation is correct for clickhouse. Implementation is used for POST comments API and create comments trpc procedure.

Reply:
LGTM